cmd/atlas/internal/cmdapi/migrate.go

```diff
@@ -716,6 +720,29 @@ func to(ctx context.Context, client *sqlclient.Client) (migrate.StateReader, err
 	}
 }
 
+// selectScheme validates the scheme of the provided to urls and returns the selected
+// url scheme. Currently, all URLs must be of the same scheme, and only multiple
+// "file://" URLs are allowed.
+func selectScheme(urls []string) (string, error) {
+	var previous string
+	if len(urls) == 0 {
+		return "", errors.New("at least one to url is required")
+	}
+	for _, url := range urls {
+		parts := strings.SplitN(url, "://", 2)
+		switch current := parts[0]; {
+		case previous == "":
+			previous = current
+			continue
```

Comment:
```suggestion
```

previous/current is misleading here. You have one scheme and it's the first one (it's not updated).